pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -0,0 +1,269 @@
+	}
+
+	if hasWildcard(rule.Verbs) && len(rule.Verbs) > 1 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("verbs"), rule.Verbs, "if '*' is present, must not specify other verbs"))
+	}
+	if hasWildcard(rule.APIGroups) && len(rule.APIGroups) > 1 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("apiGroups"), rule.APIGroups, "if '*' is present, must not specify other api groups"))
+	}
+	if hasWildcard(rule.Resources) && len(rule.Resources) > 1 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("resources"), rule.Resources, "if '*' is present, must not specify other resources"))
+	}
+	if !supportedVerbs.IsSuperset(sets.NewString(rule.Verbs...)) {
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("verbs"), rule.Verbs, supportedVerbs.List()))
+	}
+
+	return allErrs
+}
+
+func ValidateFlowSchemaStatus(status *flowcontrol.FlowSchemaStatus, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	// conditions will not be validated
+	return allErrs
+}
+
+func ValidatePriorityLevelConfiguration(pl *flowcontrol.PriorityLevelConfiguration) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&pl.ObjectMeta, false, ValidatePriorityLevelConfigurationName, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidatePriorityLevelConfigurationSpec(&pl.Spec, pl.Name, field.NewPath("spec"))...)
+	allErrs = append(allErrs, ValidatePriorityLevelConfigurationStatus(&pl.Status, field.NewPath("status"))...)
+	return allErrs
+}
+
+func ValidatePriorityLevelConfigurationSpec(spec *flowcontrol.PriorityLevelConfigurationSpec, name string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if name != flowcontrol.PriorityLevelConfigurationNameSystemTop && spec.Exempt {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("exempt"), "must not be exempt"))
+	}
+	if name != flowcontrol.PriorityLevelConfigurationNameWorkloadLow && spec.GlobalDefault {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("globalDefault"), "must not be global default"))
+	}
+	if !spec.Exempt {
+		if spec.AssuredConcurrencyShares <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("assuredConcurrencyShares"), spec.AssuredConcurrencyShares, "must be positive"))
+		}
+		if spec.QueueLengthLimit <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("queueLengthLimit"), spec.QueueLengthLimit, "must be positive"))
+		}
+		if spec.Queues <= 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("queues"), spec.Queues, "must be positive"))
+		}
+		if spec.HandSize < 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("handSize"), spec.HandSize, "must be positive"))
+		}
+	} else {
+		if spec.AssuredConcurrencyShares != 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("assuredConcurrencyShares"), spec.AssuredConcurrencyShares, "must be positive for exempt priority"))
```

Comment:
"Must be positive" is not the right complaint here; in this case the field should be "omitted" or "absent" or "zero".  Perhaps better is to use `field.Forbidden` rather than `field.Invalid`.  See https://github.com/kubernetes/kubernetes/blob/v1.15.2/pkg/apis/core/validation/validation.go#L425 for an example.

Reply:
updated to forbidden